source/developers-guide/debugging/index.md

```diff
@@ -0,0 +1,209 @@
+are putting it to, is actually rendered. 
+
+In this example the `{debug}` tag was put to the file `themes/Frontend/Bare/frontend/index/index.tpl` into the block `frontend_index_html`.
+
+```
+{block name='frontend_index_html'}
+{debug}
+// rest of the block
+{/block}
+```
+
+After clearing the cache and refreshing the page, smarty will generate a new window like this:
+
+<img style="margin-right:10px" src="img/smarty.png" />
+
+As you can see, you have a nice overview of all variables and assignments.
+
+### Debug-Plugin
+Shopware also ships with a plugin called "debug" which will allow you to print out template assignments to the `console` tab of your developer tools window.
+Just install the plugin using shopware's plugin manager, configure it to your needs and reload the page.
+
+As you can restrict the plugin to your own IP, this is also suitable for production environments.
+
+![debug plugin in action](img/debug_plugin.png)
+
+## ExtJS
+Debugging ExtJS errors during development can be very time consuming. Errors like `c is not a constructor` are not helpful
+in many cases. For that reason you can include `ext-all-debug.js` instead of the default `ext-all.js` file:
+Edit the file `themes/Backend/ExtJs/backend/base/header.tpl` and replace `ext-all.js` with `ext-all-debug.js` in the block `backend/base/header/javascript`. 
+After clearing the cache and reloading the backend, the "debug" ExtJS file is included which supports more speaking error messages.
+
+In many cases you will be no way around to debug the error by hand by using the `console.log()` call at certain points. This will help you
+to narrow down the error. The following list should help you doing that:
+
+Common backend development mistakes:
+ 
+ * Invalid class names: The name of your ExtJS class (in the `define` call) must match your directory path. E.g. `Views/backend/my_plugin/view/window` 
+  should become `Shopware.apps.MyPlugin.view.Window`
+ * Referencing a wrong xtype: Whenever you use `xtype` to reference a ExtJS class, you should double check, if the referenced xtype actually exists.
+ * Not registering the components: As ExtJS must actually know your components, you need to either register them in the `app.js` file or (when extending
+ pre-existing modules) include them using smarty and extending the original original applications `app.js` block.
+ * Missing call to `callParent(arguments);`: When implementing own components in ExtJS, you will overwrite base-components
+ a lot. Whenever you are implementing a constructor like `initComponent` or `init` you should call `callParent(arguments);`
+ so that ExtJS can handle the base component's logic.
+ * Smarty errors: Remember that  smarty parses the javascript backend files. For that reason, javascript objects always need
+ to have whitespaces before and after the opening and closing curly brace. This also applies for your comments! So if your IDE 
+  generates a DocBlock like this:
+      
+    ```
```

Comment:
Superfluous whitespace  in example code. Please remove the additional indentation.